extensions/eclipse-che-theia-credentials/src/node/che-credentials-server.ts

```diff
@@ -0,0 +1,97 @@
+/**********************************************************************
+ * Copyright (c) 2021 Red Hat, Inc.
+ *
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ ***********************************************************************/
+
+import * as k8s from '@kubernetes/client-node';
+
+import { CredentialsServer, PasswordContent } from '../common/credentials-protocol';
+import { inject, injectable } from 'inversify';
+
+import { CheK8SServiceImpl } from '@eclipse-che/theia-remote-impl-che-server/lib/node/che-server-k8s-service-impl';
+import { CheServerWorkspaceServiceImpl } from '@eclipse-che/theia-remote-impl-che-server/lib/node/che-server-workspace-service-impl';
+
+@injectable()
+export class CheCredentialsServer implements CredentialsServer {
```

Comment:
For this classe, we should rely on the mounted files for reading secrets.

About delete and create it should be possible to do it using K8S API

Reply:
> About delete and create it should be possible to do it using K8S API

Could you please elaborate more on this?